### ISSUE TITLE
Pull request expiration

### ISSUE THREAD

**Issue by scottcorgan** [desiringgod/guides#30]
I'd like to propose that all PRs have an expiration of minimum 7 days and maximum 13 days. A minimum and maximum because we would _trim_ the PR list every Monday morning.

If the PR is `>= 7 days old`, it would be closed. This provides the following

* urgency to discuss 
* removes cruft in the PR backlog
* encourages action

To address discussions that may need more than 13 days, we can add a label titled `can't expire` to notify the _trimmer_ to pass over that PR. This can be abused so it should be used with caution and sparingly.

**Comment by joeosburn-desiringgod** [comment]
I love this idea

**Comment by isaachvazquez** [comment]
Is the maximum (13 days) needed since they'll be closed at the minimum (7 days) anyway?

**Comment by benhutton** [comment]
@scottcorgan I assume you're talking about PRs everywhere, and not just this repo?

I'm feeling pretty ambivalent about this.

What is the problem you are trying to solve here?
 - Too many open PRs.
 - PRs are getting stale
 - PRs are taking too long to get reviewed
 - PRs are getting opened too early?
 - PRs are too big — they're too monolithic
 - We could go on....

I think maybe I would like to see a set of _positive_ guides, describing best practices to manage PR life cycle. At least, do that first. And then bring the hammer down if people aren't working within the new guidelines.

**Comment by scottcorgan** [comment]
@benhutton I did mean only this repo.

It was something I've done for a long time in other orgs and it helped keep the guides up to date and encouraged anyone who cared to not forget to give their input.

The main problem it's trying to solve is to keep iterating. I should probably have made it more clear that it's 7-13 days after the last activity on the PR, not since the PR was opened.

**Comment by scottcorgan** [comment]
I updated the description to reflect that

**Comment by joeosburn-desiringgod** [comment]
FWIW, I thought this was in regards to all repos.

**Comment by benhutton** [comment]
@scottcorgan in that case, I'm a fan :+1:

**Comment by scottcorgan** [comment]
@joeosburn do you think it's helpful outside this repo?

**Comment by joeosburn-desiringgod** [comment]
@scottcorgan Yes, I don't like seeing PRs flounder and not get addressed. Also, generally, I like order and directives.

**Comment by joeosburn-desiringgod** [comment]
And I like things getting done.

**Comment by shanebo** [comment]
I like @benhutton's questions and I tend to identify with @joeosburn's opinion. When I have an open PR that's waiting review it's hard for me to move on to something else until that's completed — call it a character flaw. Anyway, if we employed something like this for all PRs what would notify us if it's getting close to deadline?

**Comment by scottcorgan** [comment]
@shanebo that's a simple script to write and host on something like a free tier of Heroku or Webtask (if we actually wanted that notification. Also, there might already be a slack bought we can configure to do that.

It would be less of a notification to the author and more the job of the _trimmer_ to just close the issues after a max of 13 days. The author would get a notification at that point any way if they needed to contest it. No need to over complicate it.

**Comment by chrismaddocks** [comment]
Old issue, but since it's still _open_ I will offer an opinion...

I'm in favour of this for `guides`. I'm not in favour of having my PRs auto-expire on code repos. I _would_ be in favour of some kind of PR _deadline_ à la #36. I think I have the same character flaw as @shanebo and don't enjoy having PRs languish. 

Having my PRs auto-expire because my reviewer got busy seems counter-productive. I think the helpful question to ask here is: what % of PRs are we submitting that we ultimately expect to get merged (versus PRs we're just submitting for discussion or POC)? My sense is we're heavily weighted towards the former, so the default behaviour should reflect that, and be to keep PRs open until to merged.
